.github/workflows/cpu-packages.yml

```diff
@@ -30,22 +34,28 @@ jobs:
           sudo apt-get install -y protobuf-compiler
       - name: Install and upgrade python packages
         run: |
-          python -m pip install --upgrade pip setuptools==59.4.0 wheel tox pybind11
+          python -m pip install --upgrade pip setuptools==59.4.0 wheel pybind11
           python -m pip uninstall protobuf -y
           python -m pip install --no-binary=protobuf protobuf
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
       - name: Generate package for conda
         id: conda_build
         run: |
+          echo "conda pkgs dir $CONDA_PKGS_DIRS"
           conda update conda
-          conda install conda-build pybind11
-          conda build --python ${{ matrix.python-version }} . -c defaults -c conda-forge -c numba -c rapidsai -c nvidia --output-folder ./conda_packages
-          export CONDA_PACKAGE=$(conda build --python ${{ matrix.python-version }} . -c defaults -c conda-forge -c numba -c rapidsai -c nvidia --output-folder ./conda_packages --output)
-          echo "conda_package : $CONDA_PACKAGE"
-          echo "conda_package=$CONDA_PACKAGE" >> $GITHUB_OUTPUT
+          conda install -c conda-forge mamba
+          mamba install -c conda-forge conda-build boa conda-verify pybind11
+          conda mambabuild --python ${{ matrix.python-version }} . -c defaults -c conda-forge -c numba -c rapidsai -c nvidia --output-folder ./conda_packages
```

Comment:
Using mambabuild instead of regular conda build cut the build time down from 3-5 hours to about 10 minutes.